### PR TITLE
fix JSON marshall for floats in messages

### PR DESCRIPTION
**Pull request by mmaxim**

### CODE REVIEW

File: go/protocol/chat1/extras.go [hunk 1]
```diff
@@ -3034,6 +3036,34 @@ func (c Coordinate) Eq(o Coordinate) bool {
...
+	if math.IsNaN(lat) {
+		lat = 0
+	}
+	if math.IsNaN(lon) {
+		lon = 0
+	}
+	if math.IsNaN(accuracy) {
+		accuracy = 0
+	}
+	mlat, err := json.Marshal(lat)
```
Comment (joshblum) [inline go/protocol/chat1/extras.go:3052]: why not just have a struct with the three fields?

File: go/protocol/chat1/extras.go [hunk 2]
```diff
@@ -3153,6 +3183,30 @@ func (m AssetMetadata) IsType(typ AssetMetadataType) bool {
 	return mtyp == typ
 }
 
+func (m AssetMetadataImage) MarshalJSON() ([]byte, error) {
+	amps := make([]float64, 0, len(m.AudioAmps))
+	for _, amp := range m.AudioAmps {
+		if math.IsNaN(amp) {
```
Comment (joshblum) [inline go/protocol/chat1/extras.go:3189]: can you just replace at the indices ? why do we need to copy to a new array?

File: go/protocol/chat1/extras.go [hunk 3]
```diff
@@ -3153,6 +3183,30 @@ func (m AssetMetadata) IsType(typ AssetMetadataType) bool {
...
+	}
+	height, err := json.Marshal(m.Height)
+	if err != nil {
+		return nil, err
+	}
+	mamps, err := json.Marshal(amps)
+	if err != nil {
+		return nil, err
+	}
+	return []byte(fmt.Sprintf(`{"width":%s,"height":%s,"audioAmps":%s}`, width, height, mamps)), nil
```
Comment (joshblum) [inline go/protocol/chat1/extras.go:3207]: why not have a single struct here too

Comment (mmaxim) [reply go/protocol/chat1/extras.go:3207]: Yeah I thought about it but felt bad about replicating the struct, but I guess that is basically done anyway.

**Pull request merged**